File: 3738-make-array-non-decreasing/3738-make-array-non-decreasing.py
class Solution(object):
    def maximumPossibleSize(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        cnt = 1
        num = nums[0]
        for i in range(1, len(nums)):
            if nums[i] >= num:
                cnt += 1
                num = nums[i]
        return cnt

File: 3738-make-array-non-decreasing/3738-make-array-non-decreasing.java
class Solution {
    public int maximumPossibleSize(int[] nums) {
        int n = nums.length, cnt = 1;
        int num = nums[0];
        for (int i = 1; i < n; i++) {
            if (nums[i] >= num) {
                cnt++;
                num = nums[i];
            }
        }
        return cnt;
    }
}